src/main/java/bit/javaoop/managersPackage/Manager.java

```diff
@@ -0,0 +1,33 @@
+package bit.javaoop.managersPackage;
+
+import bit.javaoop.employeesPackage.AbstractEmployee;
+import bit.javaoop.employeesPackage.Employee;
+
+import java.util.ArrayList;
+
+
+public abstract class Manager extends AbstractEmployee {
+    protected double budget;
+    protected double employeesLimit;
+    private double budgetLeft = budget;
+
+    public Manager(String name, double salary, double budget, double employeesLimit) {
+        super(name, salary);
+        this.budget = budget;
+        this.employeesLimit = employeesLimit;
+    }
+
+    public static ArrayList<Employee> employeesList;
+
+
+    public abstract boolean canHireEmployee();
+
+    public void hireEmployee(Employee employee) {
+        employeesList.add(employee);
+        budgetLeft -= employee.getSalary();
+        employeesLimit -= 1;
+        Employee employee2 = new Employee(employee.getName(), employee.getSalary());
```

Comment:
no właśnie tutaj jakby traktujesz tę klasę jako sumę obu rodzajó managerów taką klasę która posiada cechy i tego i tego a powinna to być część wspólna zajmująca się kodem/odpowiedzialnościami które są takie same u obu managerów.